File: Programmers/Level2/구명보트.py
def solution(people, limit):
    answer = len(people)
    people = sorted(people, reverse = True)
    s,e = 0, len(people)-1
    while s < e : 
        if people[s]+people[e] <= limit :
            e-=1
            answer-=1
        s+=1
    return answer

people = [70, 50, 80, 50]
limit = 100

print(solution(people, limit))

File: Programmers/Level2/구명보트/구명보트.js
function solution(people, limit) {
  let answer = 0;
  people = people.sort((a, b) => b - a);

  for (let i = 0, j = people.length - 1; i <= j; i++, answer++)
    if (people[i] + people[j] <= limit) j--;

  return answer;
}

const people = [70, 50, 80, 50];
const limit = 100;
console.log(solution(people, limit));
